src/resolve/resolved-manager.c

```diff
@@ -1200,7 +1200,7 @@ int manager_compile_dns_servers(Manager *m, OrderedSet **dns) {
         return 0;
 }
 
-int manager_compile_search_domains(Manager *m, OrderedSet **domains, int filter_route) {
+int manager_compile_search_domains(Manager *m, OrderedSet **domains, bool filter_route) {
```

Comment:
Its a common pattern in our code to use "int"s as "bool"s with an additonal "undefined" state. See config_parse_tristate for example. I think its's a pretty ok pattern and natural extension of traditional C booleans which used to be stored as integers. Moreover real C99 bools cast implicitly and correctly into these "tristate" bools which also makes them nice to use.

Hence i think the tristate concept is fine. but of course we can remove the whole thing entirely here as pointed out.